### ISSUE TITLE
Can't run whitewidow

### ISSUE THREAD

**Issue by noptrix** [Ekultek/whitewidow#55]
## Issue/Enhancement/Question (be specific)
Can't run whitewidow due to ruby (module) related error

## Exact error message/Enhancement information
```
$ whitewidow
/usr/share/whitewidow/vendor/bundle/ruby/2.4.0/gems/rest-client-2.0.0/lib/restclient/request.rb:198:in `fetch': key not found: :ciphers (KeyError)
        from /usr/share/whitewidow/vendor/bundle/ruby/2.4.0/gems/rest-client-2.0.0/lib/restclient/request.rb:198:in `initialize'
        from /usr/share/whitewidow/vendor/bundle/ruby/2.4.0/gems/rest-client-2.0.0/lib/restclient/request.rb:52:in `new'
        from /usr/share/whitewidow/vendor/bundle/ruby/2.4.0/gems/rest-client-2.0.0/lib/restclient/request.rb:52:in `execute'
        from /usr/share/whitewidow/lib/modules/core/settings.rb:9:in `page'
        from /usr/share/whitewidow/lib/modules/core/settings.rb:18:in `parse'
        from /usr/share/whitewidow/lib/misc/banner.rb:36:in `version_type'
        from /usr/share/whitewidow/lib/imports/constants_and_requires.rb:82:in `<top (required)>'
        from whitewidow.rb:2:in `require_relative'
        from whitewidow.rb:2:in `<main>'
```

## Steps to reproduce if applicable or steps on what should be done
whitewidow

## Whitewidow version number (must have the actual version run `ruby whitewidow.rb --version`)
git master branch

## Ruby version number (run ruby --version)
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]

## Labels

 - [ ] Bug
 - [ ] Encoding
 - [ ] User agent
 - [ ] Enhancement
 - [x] Question
 - [ ] Email request

**Comment by Ekultek** [comment]
Currently trying to reproduce issue. Please provide your operating system

**Comment by Ekultek** [comment]
Unable to reproduce this issue, did you `chmod` the file before running it? Did you `bundle install` before running it? Please provide exact details on how you got this error, thank you.

I will close this issue in 48 hours if no further information is given

**Comment by noptrix** [comment]
thanks for your quick reply @Ekultek. so, i've got no issues with ruby-2.3. but this issue happenes with ruby-2.4.  and to your question about details: this is the way it gets packaged on blackarch: https://github.com/BlackArch/blackarch/blob/683ba7e14b1f5b76af6ce53ca6a44505874be4a7/packages/whitewidow/PKGBUILD and this is the way it does post_installation routines (bundle install, etc.): https://github.com/BlackArch/blackarch/blob/683ba7e14b1f5b76af6ce53ca6a44505874be4a7/packages/whitewidow/whitewidow.install . I've replaced ruby with ruby-2.3 as with 2.3 there are no issues as i said.

**Comment by Ekultek** [comment]
Seems that BlackArch needs to update the whitewidow source, I'll get into contact with them about it.

**Comment by Ekultek** [comment]
Made an issue on their github https://github.com/BlackArch/blackarch/issues/1625

I'll leave this open until that one is closed.

**Comment by noptrix** [comment]
@Ekultek thanks.i am main dev @ blackarch. just saw your github issue. thanks for that. we are using your git master branch and built directly (today) from: https://github.com/WhitewidowScanner/whitewidow.git - i am gonna switch to release tags. thanks for your help!

**Comment by Ekultek** [comment]
No problem man, good luck and thanks for using my tool :)

**Comment by Ekultek** [comment]
@noptrix I'd suggest if you can, pull it directly from Git during installation, that way it's as updated as possible. I've been working around the new issues that have arisen from the update to Ruby, so keep in touch and let me know if you find anything else, thank you

**Comment by Ekultek** [comment]
Also let's keep this open, I have a hunch that this isn't gonna solve the issue

**Comment by Ekultek** [comment]
@noptrix got a question for you, what happens if you change the depends on the PKGBUILD file from

`depends=('ruby2.3' 'ruby2.3-bundler')`
To
`depends=('ruby2.4' 'ruby2.4-bundler')` ?

**Comment by noptrix** [comment]
@Ekultek sure, you are welcome. yeah keep the issue open then. to your question: it was 'ruby' and 'ruby-bundler' before which stands for the 'newest' ruby version (e.g. 2.4) in archlinux. this wasn't working so i am forcing to use ruby-2.3 for whitewidow.

**Comment by Ekultek** [comment]
@noptrix I have a possible idea for you. What if you downloaded it by the latest tarball, and immediately upgraded it with the `--upgrade` flag during initialization of the operating system? I've come across some info on `nokogiri` and `2.4` they had to release a new patch for the gem due to the release of `2.4` so this might have something to do with that gem

**Comment by noptrix** [comment]
@Ekultek I can switch to ruby-2.4 again and do what you said. I will let you know.

**Comment by Ekultek** [comment]
@noptrix any updates on this?

**Comment by noptrix** [comment]
@Ekultek sorry, busy at the moment. i will give a try tomorrow and ping you then :)

**Comment by Ekultek** [comment]
All good sounds like a plan, thanks

**Comment by Ekultek** [comment]
@noptrix Any updates on this?

**Comment by noptrix** [comment]
@Ekultek hey, sorry for the delay. i am also not at home... so i've tried to switch back to ruby-2.4 and directly use --upgrade flag after installation with the newest release version. but this didn't worked out for me, i am still getting the same error from above. but we can keep using ruby-2.3 for now, this is not a problem at all, as ruby-2.3 is still maintained in arch linux repository.

**Comment by Ekultek** [comment]
All good man. Alright well that will work temporarily, I'd like to get a permanent fix for it though so I'll figure something out 

- Thomas Perkins

**Comment by noptrix** [comment]
@Ekultek kewl, thanks a lot for your support and follow-up.

**Comment by Ekultek** [comment]
@noptrix been doing research and it turns out this is a restclient issue, looking into it

**Comment by Ekultek** [comment]
Made an issue on their github, let's see if they have any idea. https://github.com/rest-client/rest-client/issues/600

**Comment by noptrix** [comment]
@Ekultek Awesome, thanks for the follow-up! Yep, let's see what they say...

**Comment by Ekultek** [comment]
Should be fixed via 950a43b, I'll keep this open so you can test it out.

**Comment by smtabakoglu** [comment]
Hi @Ekultek 
I have this error 
"opt/whitewidow/lib/imports/constants_and_requires.rb:6:in `require': cannot load such file -- rest-client (LoadError)
	from /opt/whitewidow/lib/imports/constants_and_requires.rb:6:in `<top (required)>'
	from whitewidow.rb:2:in `require_relative'
	from whitewidow.rb:2:in `<main>'
"
And I tried to install again rest-client

"gem install rest-client
Successfully installed rest-client-2.0.2
Parsing documentation for rest-client-2.0.2
Done installing documentation for rest-client after 0 seconds
1 gem installed"

Additional 
ruby --version
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]


Any comment?

Thanks

**Comment by Ekultek** [comment]
Your issue has been migrated to #73

**Comment by Ekultek** [comment]
@noptrix fixed via 950a43b
